mlflow/pyfunc/cli.py

```diff
@@ -1,13 +1,54 @@
 from __future__ import absolute_import
 
 import sys
+import os
+import subprocess
 
 import click
 import pandas
 
-from mlflow.pyfunc import load_pyfunc, scoring_server
+from mlflow.pyfunc import load_pyfunc, scoring_server, load_model_env
 from mlflow.tracking import _get_model_log_dir
 from mlflow.utils import cli_args
+from mlflow.projects import _maybe_create_conda_env, _get_conda_env_name
+import six
+
+
+def _rerun_in_conda(conda_env_path):
+    """ Rerun CLI command inside a to-be-created conda environment."""
+    conda_env_name = _get_conda_env_name(conda_env_path)
+    _maybe_create_conda_env(conda_env_path)
+    commands = []
+    commands.append("source activate {}".format(conda_env_name))
+    commands.append(_reconstruct_command_line())
+    commandline = " && ".join(commands)
+    child = subprocess.Popen(["bash", "-c", commandline], close_fds=True)
+    exit_code = child.wait()
+    return exit_code
+
+
+def _reconstruct_command_line():
```

Comment:
Rather than implementing by observing the click parameters, can we use `sys.argv`? I think a combination of `sys.argv` with `shlex.quote` mapped over every argument should be sufficiently robust.

Advantages:
  - Don't need to handle every type of input.
  - We don't add new parameters (e.g., defaults); we more faithfully reproduce the original input.

Reply:
sys.argv works well for manual CLI test. But it won't work when testing using click.CLIRunner (sys.argv will be the cmdline running the unit-test). It probably better not to assume its always called from a real cmdline as the function can be invoked from a python function call as well.

Reply:
Got it - I think we actually _should_ expect that this should only be invoked via real command line (i.e., not as an internal Python client), otherwise rerunning the entire command within conda could have unexpected consequences. So the assumption that `sys.argv` is appropriate may be fine in real code.

If we agree on that, then the main question is how to test this code -- we could either set `sys.argv` (and unset it) in the test code, or rely on `Popen` to more closely simulate the real CLI experience. We do that [here](https://github.com/mlflow/mlflow/blob/master/tests/helper_functions.py#L25), so there is precedence.

What do you think?